crates/wysiwyg/src/composer_model/mentions.rs

```diff
@@ -119,23 +112,21 @@ where
         self.do_insert_mention(mention_node)
     }
 
-    /// Creates a new mention node then inserts the node at the cursor position. If creation fails due to
-    /// an invalid uri, it will return `ComposerUpdate::keep()`.
-    /// It adds a trailing space when the inserted mention is the last node in it's parent.
+    /// Inserts the node at the cursor position. It adds a trailing space when the inserted
+    /// mention is the last node in it's parent.
     fn do_insert_mention(
         &mut self,
-        mention_node: DomNode<S>,
+        mention_node: MentionNode<S>, // TODO can we type this as a mention node somehow
```

Comment:
```suggestion
        mention_node: MentionNode<S>,
```

Reply:
Addressed in [46f48cb](https://github.com/matrix-org/matrix-rich-text-editor/pull/719/commits/46f48cb0f9e7c353e673b4e1179333f54bd8342c)